### PR TITLE
Update stack monitoring mapping for apm-server metrics

### PR DESCRIPTION
**Pull request by carsonip**
Update stack monitoring template for .monitoring-beats-mb to include latest apm-server monitoring metrics

All stack monitoring apm-server metrics references should be intact:
```
carson@carson-elastic:~/projects/kibana$ grep --include \*.ts -r 'beats_stats.metrics.apm-server.' x-pack/plugins/monitoring | cut -d"'" -f2 | cut -d'.' -f4- | sort | uniq | wc -l
29
carson@carson-elastic:~/projects/kibana$ grep --include \*.ts -r 'beats_stats.metrics.apm-server.' x-pack/plugins/monitoring | cut -d"'" -f2 | cut -d'.' -f4- | sort | uniq | xargs -I {} grep {} /home/carson/projects/elasticsearch/x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json | wc -l
29
```

To avoid breaking stack monitoring UI, although `beat.stats.apm_server.server.response.errors.concurrency` is unused and is not present in apm-server stats, it is manually kept in the mapping. Without this fix it would be 29, 28.

Part of https://github.com/elastic/apm-server/issues/13475

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.

### CODE REVIEW

File: x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json [hunk 1]
```diff
@@ -239,213 +232,456 @@
...
                       }
                     },
-                    "decoder": {
+                    "agentcfg": {
                       "properties": {
-                        "deflate": {
+                        "elasticsearch": {
                           "properties": {
-                            "content-length": {
+                            "cache.entries.count": {
```
Comment (carsonip) [inline x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json:245]: Probably need to expand this.

File: x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json [hunk 2]
```diff
@@ -463,9 +810,6 @@
                                 "closed": {
                                   "type": "long"
                                 },
-                                "concurrency": {
```
Comment (carsonip) [inline x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json:466]: this field is used in stack monitoring, see [code](https://github.com/elastic/kibana/blob/b8c1323d5bdc921a6238003bc698bc03750c5320/x-pack/plugins/monitoring/server/lib/metrics/apm/metrics.ts). Will this break UI?

Comment (carsonip) [reply x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json:466]: in fact, this field is never emitted by apm-server. Not sure why it is there in the first place.

Comment (carsonip) [reply x-pack/plugin/core/template-resources/src/main/resources/monitoring-beats-mb.json]: added it back to unblock progress and avoid breaking UI

**Review by klacabane — APPROVED** [review]
LGTM

**Comment by carsonip** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.15|[](https://github.com/elastic/elasticsearch/pull/110687)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**